### PR TITLE
[monitoring] remove legacy implementation of CheckAlertCondition

### PR DESCRIPTION
**Pull request by steuhs**
**What this PR does, why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:

**User-visible changes in this PR**:

### CODE REVIEW

File: tools/monitoring/config/config.go [hunk 1]
```diff
@@ -90,9 +90,9 @@ func (s *SelectedConfig) CheckAlertCondition(errorPattern string, db *sql.DB) (b
 
 	row := db.QueryRow(`
```
Comment (yt3liu) [inline tools/monitoring/config/config.go]: Do you know where is `CheckAlertCondition` currently used, except in main.go for testing this function? 
I am working on the clearing alert workflow which also has a modified version of this query. I'm still in the process for testing this query. But I don't think this method is used in the alerting flow currently. 

https://github.com/yt3liu/test-infra/blob/20d63b9b65436ffd624af918eafec5c0b176155b/tools/monitoring/mysql/database.go#L189

Comment (steuhs) [reply tools/monitoring/config/config.go]: this is the legacy implementation we no longer need. I am removing it.

**Pull request merged**